### PR TITLE
feat(auth): adding user endpoint to auth against oauth (AEROGEAR-9033)

### PR DESCRIPTION
**Pull request by laurafitzgerald**
## Motivation
JIRA: https://issues.jboss.org/browse/AEROGEAR-9033

## What
Add a endpoint to GET the user which can be used to validate if the user has authenticated with oauth in the UI

## How
Added endpoint to swagger
Added a http_handler for the user resource
Added a service layer for the user resource
Added a model for the user resource
Added the route to the router

## Verification Steps
 
1. Run make serve
2. Send a GET request to the user endpoint with the following headers
 2.a X-Forwarded-User
 2.b X-Forwarded-Email
3. The returned request should be in the following format but with the values you passed

Screenshot 2019-04-09 at 09 51 01

## Checklist:

- [x] Code has been tested locally by PR requester
- [x] Changes have been successfully verified by another team member 

## Progress

- [x] Functionality
- [x] Tests
- [ ] Verify against the [UI pr ](https://github.com/aerogear/mobile-security-service/pull/141)

### CODE REVIEW

**Comment by laurafitzgerald** [comment]
@davidffrench @craicoverflow thanks for the reviews. I will address the comments in a follow on commit. I intentionally didn't add tests initially in order to wait for the review. I will add those now. :)

**Comment by laurafitzgerald** [comment]
@davidffrench @craicoverflow changes made as requested. Tests are passing locally. Could you take a look at this again when you have some time. Thanks :)

File: cmd/mobile-security-service/main.go [hunk 1]
```diff
@@ -95,6 +96,13 @@ func setupServer(e *echo.Echo, c config.Config, dbConn *sql.DB) {
 	APIRoutePrefix := c.APIRoutePrefix
 	apiGroup := e.Group(APIRoutePrefix)
 
+	// User handler setup
+	userService := user.NewService()
```
Comment (davidffrench) [inline cmd/mobile-security-service/main.go]: this is no longer required now.

File: pkg/web/user/user_http_handler.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+// GetUser returns the user if Oauth added headers are present
+func (a *httpHandler) GetUser(c echo.Context) error {
+
+	var user models.User
+
+	//these headers values will be set by the openshift oauth-proxy
+	if userNameHeader := c.Request().Header[USER_NAME_HEADER]; userNameHeader != nil && userNameHeader[0] != "" {
+		user.Username = userNameHeader[0]
+	} else {
+		return httperrors.NotFound(c, "No User Found")
```
Comment (craicoverflow) [inline pkg/web/user/user_http_handler.go]: Should this return a 401? Where does a 401 get returned to the client?

File: pkg/web/user/user_http_handler_test.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+			wantErr:  true,
+			wantCode: 404,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			//Setup
+			e := echo.New()
+			req := httptest.NewRequest(http.MethodGet, "/", nil)
+			if !tt.wantErr {
```
Comment (craicoverflow) [inline pkg/web/user/user_http_handler_test.go]: I would put these in the test table above. Something like:

```go
{
    name:     "Should return success when header provided",
    wantErr:  false,
    wantCode: 200,
    headers: &headers{
        username: "TestUser",
        email: "test@user.com"
    }
},

...
...

req.Header.Add(USER_NAME_HEADER, tt.headers.username)
req.Header.Add(USER_EMAIL_HEADER, tt.headers.email)
```

File: pkg/web/user/user_http_handler.go (hunk 2 above)
Comment (laurafitzgerald) [reply pkg/web/user/user_http_handler.go]: oauth returns a 401 as described here: https://github.com/openshift/oauth-proxy#endpoint-documentation so we don't need to do that here.

**Pull request merged**